ush/forecast_postdet.sh

```diff
@@ -500,22 +500,20 @@ MOM6_out() {
     done
   fi
 
-  # Copy restarts at the beginning/middle of the next assimilation cycle to COM for RUN=gdas|enkfgdas|enkfgfs
+  # Copy restarts in the assimilation window for RUN=gdas|enkfgdas|enkfgfs
   if [[ "${RUN}" =~ "gdas" || "${RUN}" == "enkfgfs" ]]; then
     local restart_date
-    if [[ "${DOIAU}" == "YES" ]]; then  # Copy restarts at the beginning of the next cycle from DATA to COM
-      restart_date="${next_cycle_begin}"
-    else  # Copy restarts at the middle of the next cycle from DATA to COM
-      restart_date="${next_cycle}"
-    fi
-    echo "Copying MOM6 restarts for 'RUN=${RUN}' at ${restart_date}"
-    for mom6_restart_file in "${mom6_restart_files[@]}"; do
-      restart_file="${restart_date:0:8}.${restart_date:8:2}0000.${mom6_restart_file}"
-      ${NCP} "${DATArestart}/MOM6_RESTART/${restart_file}" \
-             "${COM_OCEAN_RESTART}/${restart_file}"
+    restart_date=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} + ${restart_interval} hours" +%Y%m%d%H)
+    while (( restart_date < forecast_end_cycle )); do
+      echo "Copying MOM6 restarts for 'RUN=${RUN}' at ${restart_date}"
+      for mom6_restart_file in "${mom6_restart_files[@]}"; do
+	restart_file="${restart_date:0:8}.${restart_date:8:2}0000.${mom6_restart_file}"
+        ${NCP} "${DATArestart}/MOM6_RESTART/${restart_file}" \
+               "${COM_OCEAN_RESTART}/${restart_file}"
+      done
+      restart_date=$(date --utc -d "${restart_date:0:8} ${restart_date:8:2} + ${restart_interval} hours" +%Y%m%d%H)
```

Comment:
If I understand this correctly, `COM` will contain 2 pairs of restarts; one at the beginning and another at the end of the assimilation window regardless of `DOIAU`.
Is there a way to separate that, such that only the neccessary restart is copied out to `COM`.
If `DOIAU=NO`, it will always be the middle of the window.
If `DOIAU=YES`, it will always be the beginning of the window.

I can see the previous logic was deficient in handling `DOIAU=YES` for the experiment that has a cold-start half cycle.

Reply:
I'm not sure what another solution would be.  Is there a variable that exists that indicates that it's the first half cycle?  What is here is using what the atmosphere model is doing for gdas, but if there's another solution I can implement that here instead.

Reply:
JEDI-based atmosphere DA requires restarts in the assimilation window, and that is why they are being copied to `COM`. 

A variable to indicate the first half cycle is a good idea to explore.

Reply:
What about: 

https://github.com/NOAA-EMC/global-workflow/blob/develop/ush/forecast_det.sh#L30-L32 These lines: 
```
    # Since warm start is false, we cannot do IAU
    DOIAU="NO"
    IAU_OFFSET=0
```
to 
```
    # Since warm start is false, we cannot do IAU - but set a variable to know the next cycle will have IAU
    if [[ "${DOIAU}" == "YES" ]]; then 
          NEXTCYCIAU=YES
    else 
          NEXTCYCIAU=NO 
    fi 
    DOIAU="NO"
    IAU_OFFSET=0
```
And then we can use "NEXTCYCIAU" in the copying to know which restart to copy?

Reply:
I slightly updated this, but am using a "NEXTCYCIAU trigger for which restarts to copy for CICE, MOM6 and CMEPS (someday can be used for ww3 too).    This would reduce the amount of restarts copied.  I'll push updates after testing.  I've marked the PR as draft to resolve this issue.

Reply:
@aerorahul trying something to see if it works.  Let me know what you think, we can try something else.  This was inspired by your ideas you shared offline, but might not have been exactly what you were originally thinking so we can tweak as needed.